phonenumber_field/widgets.py

```diff
@@ -58,22 +68,30 @@ def __init__(self, attrs=None, initial=None):
         super().__init__(widgets, attrs)
 
     def decompress(self, value):
-        if value:
-            if type(value) == PhoneNumber:
-                if value.country_code and value.national_number:
-                    return [
-                        "+%d" % value.country_code,
-                        national_significant_number(value),
-                    ]
-            else:
-                return value.split(".")
-        return [None, ""]
+        region = getattr(settings, "PHONENUMBER_DEFAULT_REGION", None)
+
+        if isinstance(value, str):
+            value = to_python(value=value, region=region)
+
+        if isinstance(value, PhoneNumber):
+            if not value.is_valid():
+                region_code = getattr(value, "_region", region)
+                return [region_code, value.raw_input]
+            region_code = region_code_for_number(value)
+            national_number = str(value.national_number)
+            return [region_code, national_number]
+        return [None, None]
 
     def value_from_datadict(self, data, files, name):
-        values = super().value_from_datadict(data, files, name)
-        if all(values):
-            return "%s.%s" % tuple(values)
-        return ""
+        region_code, national_number = super().value_from_datadict(data, files, name)
+
+        number = to_python(value=national_number, region=region_code)
```

Comment:
The value is a positional argument.
```suggestion
        number = to_python(national_number, region=region_code)
```